daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/ElementBase.scala

```diff
@@ -985,6 +985,7 @@ trait ElementBase
       "Facets minLength and maxLength are allowed only on types string and hexBinary.",
     )
     typeDef match {
+      case _ if hasRepType => (repTypeElementDecl.minLength, repTypeElementDecl.maxLength)
```

Comment:
With this change, it's clearly computing the min/max length of the representation. 

Add that to the scaladoc.

But is this correct? Is this only used when we need the max/min length of the representation? 
What about limited validation? Then we're dealing with the infoset value, and the max/min length are those of the logical data, not the rep.

Perhaps we need to distinguish infosetMinLength and repMinLength (similarly for max) ?

Reply:
This change is basically the same as before, the case just changes a little bit (see the line removed below), so functionally nothing was really changed. And it looks like things to work as expected.

If I add a min/maxLength restriction to  the logical type, then limited validation does correctly check the length of the calculated enumeration string. Note that it seems unlikely that schemas will use both min/maxLength and enums on a restriction, but if they to it seems to work. Maybe this min/maxLength caclulation is not what is used for facet checking?

Also, if I add min/maxLength to the repType simple type, then it erros. min/maxLength facets are only valid on xs:string types, and repType must be a string type.

I'll add some tests to make sure this works as expected and isn't broken in the future.